src/views/dashboard/components/communityList.js

```diff
@@ -107,36 +107,40 @@ class CommunityList extends React.Component<Props> {
           </CommunityListItem>
           {sortedCommunities.map(c => (
             <ErrorBoundary fallbackComponent={null} key={c.id}>
-              <CommunityListItem
-                onClick={() => this.handleOnClick(c.id)}
-                active={c.id === activeCommunity}
-              >
-                <CommunityAvatarContainer>
-                  <CommunityAvatar community={c} showHoverProfile={false} />
-                </CommunityAvatarContainer>
-                <CommunityListMeta>
-                  <CommunityListName>{c.name}</CommunityListName>
-                  <Reputation
-                    ignoreClick
-                    size={'mini'}
-                    tipText={`Your rep in ${c.name}`}
-                    reputation={c.communityPermissions.reputation}
-                  />
-                </CommunityListMeta>
-
-                {c.id === activeCommunity && (
-                  <ErrorBoundary>
-                    <SidebarChannels
-                      activeChannel={activeChannel}
-                      communitySlug={c.slug}
-                      permissions={c.communityPermissions}
-                      slug={c.slug}
-                      id={c.id}
-                      setActiveChannelObject={this.props.setActiveChannelObject}
+              <Link to={`/${c.slug}`}>
```

Comment:
We do not want to redirect to the community once the JS has rehydrated. I think to fix that, we need to move the `onClick` to the `Link` here!

```suggestion
              <Link to={`/${c.slug}`} onClick={() => this.handleOnClick(c.id)}>
```

Reply:
oh yes. And adding `e.preventDefault` in `onClick` handler might satisfy that condition.

Thanks.